File: term4/prog eng/lab3/Actions.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Actions.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
#include "Main.h"
#include "Output.h"

TForm4 *Form4;
//---------------------------------------------------------------------------
__fastcall TForm4::TForm4(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm4::Button1Click(TObject *Sender)
{
	float answer = (Form2->matrix).getCoff();
	(Form3->Label1)->Caption = "Coff: " + FloatToStr(answer);
}
//---------------------------------------------------------------------------
void __fastcall TForm4::Button2Click(TObject *Sender)
{
	int nulls = (Form2->matrix).getEvenRowsNulls();
	int selection = 0;
	if (!nulls) {
		selection = -1;
	}
	(Form3->RG1)->ItemIndex = selection;
}
//---------------------------------------------------------------------------
void __fastcall TForm4::FormClose(TObject *Sender, TCloseAction &Action)
{
    Action = caFree;
}
//---------------------------------------------------------------------------

File: term4/prog eng/lab5/Actions.pas
unit Actions;

interface

uses
  Winapi.Windows, Winapi.Messages, System.SysUtils, System.Variants, System.Classes, Vcl.Graphics,
  Vcl.Controls, Vcl.Forms, Vcl.Dialogs, Vcl.StdCtrls, Output, Matrix;

type
  TForm6 = class(TForm)
    Button1: TButton;
    Button2: TButton;
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form6: TForm6;

implementation

{$R *.dfm}

procedure TForm6.Button1Click(Sender: TObject);
var
  answer: Single;
begin
  answer := Matrix.AMatrix.GetCoff;
  Form4.Label1.Caption := 'Coff: ' + FloatToStr(answer);
end;



procedure TForm6.Button2Click(Sender: TObject);
var
  nulls: Integer;
  selection: Boolean;
begin
  nulls := Matrix.AMatrix.getEvenRowsNulls;
  selection := True;
  if nulls = 0 then
    selection := False;
  Form4.RadioButton1.Checked := selection;
end;

end.
